index.bs

```diff
@@ -1701,6 +1728,43 @@ a [=moment=] |sourceTime|, and a [=duration=] |eventReportWindow|:
     1. Set |lastEnd| to |lastEnd| + |deadline|.
 1. Return |windows|.
 
+To <dfn>parse report windows</dfn> given a [=map=] |map|, a [=moment=] |sourceTime|,
+and a [=event-level report window list=] |default|:
+
+1. [=Assert=]: |default| is not [=list/is empty|empty=].
+1. Let |defaultEndDuration| be the [=duration from=] |sourceTime| and the [=event-level report window/end=] of the last element of |default|.
+1. If |map|["`event_report_windows`"] does not [=map/exist=], return |default|.
+1. Let |values| be |map|["`event_report_windows`"].
+1. If |values| is not a [=map=], return an error.
+1. Let |start| be the result of running [=parse an optional 64-bit signed integer=] with |values|, "`start_time`", and 0.
+1. If |start| is an error, return an error.
+1. If |start| is less than 0, return an error.
+1. Let |startDuration| be |start| seconds.
+1. If |startDuration| is greater than |defaultEndDuration|, return an error.
+1. If |values|["`end_times`"] does not [=map/exist=] or is not a [=list=], return an error.
+1. Let |endDurations| be |values|["`end_times`"].
+1. If the [=list/size=] of |endDurations| is greater than [=max settable event-level report windows=], return an error.
+1. Let |windows| be an [=list/is empty|empty=] [=list=].
+1. [=list/iterate|For each=] |elt| of |endDurations|:
+    1. If |elt| is not a [=string=], return an error.
+    1. Let |end| be the result of applying the <a spec="html">rules for parsing integers</a> to |elt|.
+    1. If |end| is an error, return an error.
+    1. Let |endDuration| be |end| seconds.
+    1. If |endDuration| is greater than |defaultEndDuration|, set |endDuration| to |defaultEndDuration|.
+    1. If |endDuration| less than or equal to |startDuration|, return an error.
+    1. Let |window| be a new [=event-level report window=] whose items are
+
+        : [=event-level report window/start=]
+        :: |sourceTime| + |startDuration|
+        : [=event-level report window/end=]
+        :: |sourceTime| + |endDuration|
+
+    Issue: Set a stricter minimum for |endDuration|.
+
+    1. [=list/Append=] |window| to |windows|.
+    1. Set |startDuration| to |endDuration|.
+1. Return |windows|.
```

Comment:
What is supposed to happen if `windows` is empty, e.g. because the `end_times` JSON list was empty?

Reply:
1. I added a constraint to the wrapper type that it needs at least one element.
2. I added an error if |endDurations| is empty.